skyportal/handlers/api/plan.py

```diff
@@ -0,0 +1,478 @@
+    params["config"][instrument.name]["overhead_per_exposure"] = 0.0
+
+    params["skymap"] = ""
+    params["gpstime"] = -1
+    params["outputDir"] = "output/%s" % dateobs.strftime("%Y%m%dT%H%M%S")
+    params["tilingDir"] = ""
+    params["event"] = ""
+    params["telescopes"] = [instrument.name]
+    if schedule_strategy == "catalog":
+        params["tilesType"] = "galaxy"
+        params["catalogDir"] = "tmp"  # fix me
+        params["galaxy_catalog"] = "CLU"
+        params["galaxy_grade"] = "S"
+        params["writeCatalog"] = False
+        params["catalog_n"] = 1.0
+        params["powerlaw_dist_exp"] = 1.0
+        params["doChipGaps"] = False
+    elif schedule_strategy == "tiling":
+        params["tilesType"] = "moc"
+        params["doChipGaps"] = False
+    params["scheduleType"] = schedule_type
+    params["timeallocationType"] = "powerlaw"
+    params["nside"] = 512
+    params["powerlaw_cl"] = probability
+    params["powerlaw_n"] = 1.0
+    params["powerlaw_dist_exp"] = 0.0
+
+    params["doPlots"] = False
+    params["doMovie"] = False
+    params["doObservability"] = True
+    params["do3D"] = False
+
+    params["doFootprint"] = False
+    params["footprint_ra"] = 30.0
+    params["footprint_dec"] = 60.0
+    params["footprint_radius"] = 10.0
+
+    params["airmass"] = airmass
+
+    params["doCommitDatabase"] = True
+    params["doRequestScheduler"] = False
+    params["dateobs"] = dateobs
+    params["doEvent"] = False
+    params["doSkymap"] = False
+    params["doFootprint"] = False
+    params["doDatabase"] = True
+    params["doReferences"] = doReferences
+    params["doUsePrimary"] = doUsePrimary
+    params["doBalanceExposure"] = doBalanceExposure
+    params["doSplit"] = False
+    params["doParallel"] = False
+    params["doUseCatalog"] = False
+
+    params["doMinimalTiling"] = True
+    params["doIterativeTiling"] = False
+    params["galaxies_FoV_sep"] = 1.0
+    params["doMaxTiles"] = doMaxTiles
+    params["max_nb_tiles"] = np.array([max_nb_tiles] * len(filt))
+
+    if params["doEvent"]:
+        params["skymap"], eventinfo = gwemopt.gracedb.get_event(params)
+        params["gpstime"] = eventinfo["gpstime"]
+        event_time = time.Time(params["gpstime"], format='gps', scale='utc')
+        params["dateobs"] = event_time.iso
+    elif params["doSkymap"]:
+        event_time = time.Time(params["gpstime"], format='gps', scale='utc')
+        params["dateobs"] = event_time.iso
+    elif params["doFootprint"]:
+        params["skymap"] = gwemopt.footprint.get_skymap(params)
+        event_time = time.Time(params["gpstime"], format='gps', scale='utc')
+        params["dateobs"] = event_time.iso
+    elif params["doDatabase"]:
+        event_time = time.Time(params["dateobs"], format='datetime', scale='utc')
+        params["gpstime"] = event_time.gps
+    else:
+        raise ValueError(
+            'Need to enable --doEvent, --doFootprint, ' '--doSkymap, or --doDatabase'
```

Comment:
Do you have control over these flags?  If so, you may rather want something like `--action=event --action=footprint --action=database` or similar.